census_view.cpp

```diff
@@ -1268,8 +1268,57 @@ void CensusView::apply_changes
         }
 }
 
-void CensusView::update_visible_columns()
-{
+bool CensusView::update_visible_columns()
+{
+    // Check if we really need to update the columns, it takes a relatively
+    // long time and is done directly in response to the user actions, so it
+    // makes a lot of sense to avoid doing it completely as long as possible.
+    bool must_update_columns = false;
+
+    auto const& all_header_names(case_parms()[0].member_names());
+    bool must_update_headers =
```

Comment:
What's the difference between `must_update_{headers,columns}`? It looks like they end up with the same value to me...

Also, I think this `||` is too long and unwieldy, I'd use `if` for it instead, i.e.
```cpp
bool must_update_columns;
if(size changed)
{
  must_update_columns = true;
}
else
{
  must_update_columns = std::equal(...);
}
```